lib/clockwork/test/manager.rb

```diff
@@ -57,21 +57,17 @@ def set_opts(opts)
       attr_reader :history
 
       def tick_loop
-        while true do
-          update_job_history
-
-          tick
-          increase_time
-
-          @total_ticks += 1
-          break if ticks_exceeded? || time_exceeded?
+        loop.with_index do |_, index|
```

Comment:
This does change the behaviour a bit; not sure if this was intentional (the PR description does not mention this).

Before this change this would work:

clock.rb:
```ruby
  # ...
  every(1.hour, "stand up") do
    puts "Stand up!"
  end
```

clock_test.rb:
```ruby
  # ...
  test "stand up" do
    time = Time.utc(2023, 4, 21)

    Clockwork::Test.run(max_ticks: 1, start_time: time, end_time: time)
    assert_equal(1, Clockwork::Test.times_run("stand up"))
  end
```

Before we'd do at least one tick if start and end date are the same. Now `#time_exceeded?` is checked at the beginning of the loop and we exit early.

Reply:
Thanks for bringing this up. I did not intend any functional change as as result of this.

Sounds like we're missing a test case!

@robinbrandt if you're able to submit a PR to modify this, I'm happy to review. Just a heads up that I'm heading on vacation starting today, so I'll likely be delayed in reviewing.

I appreciate you commenting here!

Reply:
> if you're able to submit a PR to modify this

Happy to do so!

https://github.com/kevin-j-m/clockwork-test/pull/41.

> I'm heading on vacation

Enjoy your vacation!